railties/test/generators/api_app_generator_test.rb

```diff
@@ -0,0 +1,92 @@
+require 'generators/generators_test_helper'
+require 'rails/generators/rails/app/app_generator'
+
+class ApiAppGeneratorTest < Rails::Generators::TestCase
+  include GeneratorsTestHelper
+  tests Rails::Generators::AppGenerator
+
+  arguments [destination_root, '--api']
+
+  def setup
+    Rails.application = TestApp::Application
+    super
+
+    Kernel::silence_warnings do
+      Thor::Base.shell.send(:attr_accessor, :always_force)
+      @shell = Thor::Base.shell.new
+      @shell.send(:always_force=, true)
+    end
+  end
+
+  def teardown
+    super
+    Rails.application = TestApp::Application.instance
+  end
+
+  def test_skeleton_is_created
+    run_generator
+
+    default_files.each { |path| assert_file path }
+    skipped_files.each { |path| assert_no_file path }
+  end
+
+  def test_api_modified_files
+    run_generator
+
+    assert_file "Gemfile" do |content|
+      assert_no_match(/gem 'coffee-rails'/, content)
```

Comment:
This is testing here just against the default generated Gemfile that use single quote as convention, so this is fine.